pkgs/top-level/all-packages.nix

```diff
@@ -7711,11 +7711,13 @@ with pkgs;
   # In other words, try to only use this in wrappers, and only use those
   # wrappers from the next stage.
   bintools-unwrapped = let
-    inherit (stdenv.targetPlatform) linker;
-  in     if linker == "lld"     then llvmPackages.bintools-unwrapped
-    else if linker == "cctools" then darwin.binutils-unwrapped
-    else if linker == "bfd"     then binutils-unwrapped
-    else if linker == "gold"    then binutils-unwrapped.override { enableGoldDefault = true; }
+    # We should remove the linker inherit and solely go off of bintools in the future.
+    # The linker should be specified inside package build systems (cmake, meson, etc).
+    inherit (stdenv.targetPlatform) linker bintools;
+  in     if linker == "lld" || bintools == "llvm"      then llvmPackages.bintools-unwrapped
+    else if linker == "cctools" || bintools == "apple" then darwin.binutils-unwrapped
```

Comment:
Darwin’s bintools is defined as `llvm` above. I think that is more accurate¹, but it results in the wrong bintools being picked here.

1: Everything but the linker and `ranlib` are from LLVM, excepting platform-specific tools like `lipo` and `install_name_tool`, which are broken/incomplete in LLVM.

Reply:
Doesn't nixpkgs currently pick cctools without this change? I don't want to change behaviors if it can cause a mass rebuild.

Reply:
I just checked and `legacyPackages.aarch64-darwin.bintools-unwrapped` evals to the LLVM one.